libs/ibm/langchain_ibm/chat_models.py

```diff
@@ -619,7 +620,19 @@ def validate_environment(self) -> Self:
             check_for_attribute(self.url, "url", "WATSONX_URL")
 
             if "cloud.ibm.com" in self.url.get_secret_value():
-                check_for_attribute(self.apikey, "apikey", "WATSONX_APIKEY")
+                if not self.token and not self.apikey:
+                    raise ValueError(
+                        "Did not find 'apikey' or 'token',"
+                        " please add an environment variable"
+                        " `WATSONX_APIKEY` or 'WATSONX_TOKEN' "
+                        "which contains it,"
+                        " or pass 'apikey' or 'token'"
+                        " as a named parameter."
+                    )
+                elif self.apikey:
+                    check_for_attribute(self.apikey, "apikey", "WATSONX_APIKEY")
+                elif self.token:
+                    check_for_attribute(self.token, "token", "WATSONX_TOKEN")
```

Comment:
Do we need these `elif` statements? If at least one of the `self.apikey` or `self.token` is not None then we can freely pass them both to the `APIClient` init.